tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -454,6 +455,23 @@ def __len__(self):
       raise TypeError("dataset length is unknown.")
     return length
 
+  def __evenly_divisible__(self, batch_size):
+    if not context.executing_eagerly():
+      batch_mod = math_ops.floor_mod(self.cardinality(),batch_size)
+      cond = math_ops.logical_and(math_ops.greater(self.cardinality(),0),
+                                  math_ops.equal(batch_mod,0))
+      pred = smart_cond.smart_cond(cond, lambda: constant_op.constant(True),
+                               lambda: constant_op.constant(False))
+      #TODO(bhack) improve this
+      evenly_divisible = backend.eval(pred)
```

Comment:
@aaudiber How can I evaluate this removing Keras but also pass Distributed test?